app/assets/javascripts/discourse/app/routes/signup.js

```diff
@@ -4,17 +4,45 @@ import { service } from "@ember/service";
 import DiscourseRoute from "discourse/routes/discourse";
 
 export default class SignupRoute extends DiscourseRoute {
-  @service router;
   @service siteSettings;
+  @service router;
+  @service login;
+
+  autoRedirect = false;
+  authComplete = false;
+  singleLoginMethod = false;
 
-  beforeModel() {
-    this.showCreateAccount();
+  beforeModel(transition) {
+    const loginRequired = this.siteSettings.login_required;
+    this.authComplete = transition.to.queryParams.authComplete || false;
+    this.singleLoginMethod = this.login.isOnlyOneExternalLoginMethod;
+    this.autoRedirect =
+      !this.authComplete && this.singleLoginMethod && !loginRequired;
```

Comment:
```suggestion
    this.authComplete = transition.to.queryParams.authComplete || false;
    this.singleLoginMethod = this.login.isOnlyOneExternalLoginMethod;
    this.autoRedirect =
      !this.authComplete && this.singleLoginMethod && ! this.siteSettings.login_required;
```

Tiny suggestion, since it's only being used in one place, no need to store a constant, I think.